### PR TITLE
Fix/selected text deletion bug

### PR DESCRIPTION
**Pull request by aurelienshz**
See the long comment added by this PR for detailed explanation of the pros & cons of this PR.

TL;DR: this.buffer.set() saves the current state of the editor, then restores the selection & caret position. We disable this behaviour when some text is actually selected, so that the deleted text doesn't get instantly reinserted.

### CODE REVIEW

File: vendor/assets/javascripts/redactor-rails/redactor.js [hunk 1]
```diff
@@ -1026,22 +1026,57 @@
 		buffer: function()
 		{
 			return {
-				set: function(type)
+				set: function(type, opts)
 				{
 					if (typeof type == 'undefined' || type == 'undo')
 					{
-						this.buffer.setUndo();
+						this.buffer.setUndo(opts);
```
Comment (chengdiao) [inline vendor/assets/javascripts/redactor-rails/redactor.js]: opts unused

File: vendor/assets/javascripts/redactor-rails/redactor.js [hunk 2]
```diff
@@ -4808,7 +4842,7 @@
 							if (this.opts.linebreaks)
 							{
 								$(this.keyup.current).after(this.selection.getMarkerAsHtml());
-								this.selection.restore();
+								tore();
```
Comment (chengdiao) [inline vendor/assets/javascripts/redactor-rails/redactor.js]: restore

**Review by chengdiao — COMMENTED** [review]
clean up

**Comment by aurelienshz** [comment]
Comments addressed

**Pull request merged**